src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs

```diff
@@ -114,6 +114,16 @@ void OnContentChanged(DependencyObject sender, DependencyProperty dp)
 		void OnShadowHostSizeChanged(object sender, SizeChangedEventArgs args)
 		{
 			_isShadowHostDirty = true;
+
+			// This is not necessary on WinUI, but is necessary on Uno due to behavior mismatch in ActualWidth/ActualHeight.
+			// For the ShadowContainerSamplePage, when we add a new shadow, we are expanding the StackPanel that's wrapped in ShadowContainer.
+			// We get through InvalidateCanvasLayout via OnContentSizeChanged and we update SKXamlCanvas.Height to the new Height
+			// On WinUI, the updated Height is immediately reflected on ActualHeight, but that's not the case on Uno.
+			// So, SKXamlCanvas still has its internal SKCanvas with the old size (as it relies on ActualHeight)
+			// see https://github.com/mono/SkiaSharp/blob/a290ccffc74bc95a45569e8b74e77be851d4a688/source/SkiaSharp.Views.Uno/SkiaSharp.Views.Uno.WinUI.Shared/SKXamlCanvas.cs#L140
+			// Then we end up drawing the old size.
+			// To account for Uno's mismatched behavior, we call InvalidateShadows.
+			InvalidateShadows();
```

Comment:
@Youssef1313 can we add some tests for this?

Reply:
I think we are not running tests in CI against Skia anyway. It's not reproducible on Wasm (and possible all the other platforms)

Reply:
cc @kazo0

Reply:
We should add support for Skia on the CI (https://github.com/unoplatform/uno.toolkit.ui/issues/1095)

Is there still anything here that could be covered by a test? Even if it's not yet on the CI?